### DISCUSSION TITLE
Relational / Reference item inside another collection - in a repeater

### DISCUSSION THREAD

**Discussion by sagive** [directus/directus#11181] [Archive]
Hey guys, DIrectus looks awesome! 
a user can feel the brilliancy of the team crafting it!

I have kinda a novice question I think.  i am just trying to reference a row from another collection
so that I could later use that "global" information in the front

THis is the structure I am aiming for - the last part inside the repeater is
where I am stuck... is this a feature request or am I just missing something?

**channels (collection)**
 - name
 - url
 - desc

**accounts (collection)**
 - person_name
 - email
 - phone
 - images
 - date_create
 - user_created
 - accounts_data (repeater)
    - user
    - pass
    - channel (pick am item from channels collection for later reference)   <--- how do you do that? 
    
Even as JSON I should be able to save a ref that I can later use no?
in any case, thank you guys a lot. Appreciate you 🙏👍

**Comment by rijkvanzanten** [comment]
Heya! In this case, you'd want to use a one to many (or many to many, based on your data needs) field instead of a repeater. A JSON repeater doesn't hold any relational data, as it's just a "dumb" object of static values.

**Reply by sagive** [reply]
Thanks for responding (and so fast) man, 
does that mean that there is no way to connect an item inside a repeater row to an external value?
it's not really a dynamic value. just (for example)  might be a list of categories.

Basically, there is no way to create my schema 😭
i must split it into an additional collection, load it with values and that use two relationships,
all to evade using a repeater - its a bit weird I think...

thank you for your help 🙏

**Reply by rijkvanzanten** [reply]
Based on the schema you proposed above, I would structure it as `channels`, `accounts`, and `credentials`, where each `credentials` record is associated with an account and channel.

> i must split it into an additional collection, load it with values and that use two relationships,
all to evade using a repeater - its a bit weird I think...

I disagree. By injecting the primary keys of the related items in a static json object, there's no way to search for them, filter based on which ones exist, do any useful exports, of have any relational constraints to the tokens, which in turn means there's no guarantee that the tokens actually point to a related channel.

**Reply by sagive** [reply]
The case i have shown is just one case, I would probably have to do something similar 3-6 times on other resources
which would get messy (and not in a good "Lionel Messi" kinda way 😁)

>  there's no way to search for them, filter based on which ones exist, do any useful exports, of have any relational constraints to the tokens, which in turn means there's no guarantee that the tokens actually point to a related channel.

Why, what happens if I delete some value inside a "connected" collection?
a reference to a data point is nothing more than that. It's up to the programmer to show the
data, validate it exists or offer an alternative (in my opinion). 

Nevertheless, I understand you guys made decisions while taking into account
a multitude of considerations and your personal preferences which is no less important. 

Thank you very much for the assist.
and have a gr8 weekend ❤️🙏
